src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -194,14 +206,141 @@ internal void UpdateSelectedItem(PackageItemListViewModel selectedItem)
+                    if (!Items.Contains(_loadingStatusIndicator))
+                        {
+                            Items.Add(_loadingStatusIndicator);
+                            System.Diagnostics.Debug.WriteLine("LoadItems LoadingStatus added (" + Items.Count + ")"); //TODO remove
+                        addedLoadingIndicator = true;
+                        }
+
+                        await LoadItemsCoreAsync(currentLoader, loadCts.Token);
+
+                        await _joinableTaskFactory.Value.SwitchToMainThreadAsync();
+
+                    //If we refreshed, then any filter should be cleared.
+                    if (!applyUIFilterUpdatesAvailable)
+                        {
+                            ClearItemsFilter();
+                        }
+
+                        if (selectedPackageItem != null)
+                        {
+                            UpdateSelectedItem(selectedPackageItem);
+                        }
+                    }
+                    catch (OperationCanceledException) when (!loadCts.IsCancellationRequested)
+                    {
+                        loadCts.Cancel();
+                        loadCts.Dispose();
+                        currentLoader.Reset();
+
+                        await _joinableTaskFactory.Value.SwitchToMainThreadAsync();
+
+                    // The user cancelled the login, but treat as a load error in UI
+                    // So the retry button and message is displayed
+                    // Do not log to the activity log, since it is not a NuGet error
+                    _logger.Log(ProjectManagement.MessageLevel.Error, Resx.Resources.Text_UserCanceled);
+
+                        _loadingStatusIndicator.SetError(Resx.Resources.Text_UserCanceled);
+
+
+                        _loadingStatusBar.SetCancelled();
+                        _loadingStatusBar.Visibility = Visibility.Visible;
+                    }
+                    catch (Exception ex) when (!loadCts.IsCancellationRequested)
+                    {
+                        loadCts.Cancel();
+                        loadCts.Dispose();
+                        currentLoader.Reset();
+
+                    // Write stack to activity log
+                    Mvs.ActivityLog.LogError(LogEntrySource, ex.ToString());
+
+                        await _joinableTaskFactory.Value.SwitchToMainThreadAsync();
+
+                        var errorMessage = ExceptionUtilities.DisplayMessage(ex);
+                        _logger.Log(ProjectManagement.MessageLevel.Error, errorMessage);
+
+                        _loadingStatusIndicator.SetError(errorMessage);
+
+                        _loadingStatusBar.SetError();
+                        _loadingStatusBar.Visibility = Visibility.Visible;
+                    }
+                    catch (Exception ex)
+                    {
+                        var message = ex.ToString();
+                        LoadItemsCanceled?.Invoke(this, EventArgs.Empty);
```

Comment:
Basically wanted to tell the PMControl to "stop" anything it's doing. Throwing here did not talk to the PMControl.